Build failure on Fedora 23+

Trying to build 2015.11.11 on Fedora (I'm the package maintainer) and it's failing in garbagecollector:

 33%] Linking C shared library _build/dll/libgarbagecollector.so
CMakeFiles/garbagecollector.dir/source/CollectorMarker.o: In function `List_size.localalias.6':
CollectorMarker.c:(.text+0x0): multiple definition of `List_size'
CMakeFiles/garbagecollector.dir/source/Collector.o:Collector.c:(.text+0x0): first defined here
CMakeFiles/garbagecollector.dir/source/CollectorMarker.o: In function `List_ifNeededSizeTo_.localalias.5':
CollectorMarker.c:(.text+0x12): multiple definition of `List_ifNeededSizeTo_'
CMakeFiles/garbagecollector.dir/source/Collector.o:Collector.c:(.text+0x12): first defined here
CMakeFiles/garbagecollector.dir/source/CollectorMarker.o: In function `List_rawAt_':
CollectorMarker.c:(.text+0x51): multiple definition of `List_rawAt_'
CMakeFiles/garbagecollector.dir/source/Collector.o:Collector.c:(.text+0x51): first defined here
CMakeFiles/garbagecollector.dir/source/CollectorMarker.o: In function `List_at_.localalias.4':
CollectorMarker.c:(.text+0x74): multiple definition of `List_at_'
CMakeFiles/garbagecollector.dir/source/Collector.o:Collector.c:(.text+0x74): first defined here
CMakeFiles/garbagecollector.dir/source/CollectorMarker.o: In function `List_indexOf_':
CollectorMarker.c:(.text+0xc9): multiple definition of `List_indexOf_'
CMakeFiles/garbagecollector.dir/source/Collector.o:Collector.c:(.text+0xc9): first defined here
CMakeFiles/garbagecollector.dir/source/CollectorMarker.o: In function `List_contains_':
CollectorMarker.c:(.text+0x134): multiple definition of `List_contains_'
CMakeFiles/garbagecollector.dir/source/Collector.o:Collector.c:(.text+0x134): first defined here
CMakeFiles/garbagecollector.dir/source/CollectorMarker.o: In function `List_append_.localalias.3':
CollectorMarker.c:(.text+0x19e): multiple definition of `List_append_'
CMakeFiles/garbagecollector.dir/source/Collector.o:Collector.c:(.text+0x19e): first defined here
CMakeFiles/garbagecollector.dir/source/CollectorMarker.o: In function `List_appendSeq_':
CollectorMarker.c:(.text+0x200): multiple definition of `List_appendSeq_'
CMakeFiles/garbagecollector.dir/source/Collector.o:Collector.c:(.text+0x200): first defined here
CMakeFiles/garbagecollector.dir/source/CollectorMarker.o: In function `List_compactIfNeeded.localalias.2':
CollectorMarker.c:(.text+0x26c): multiple definition of `List_compactIfNeeded'
CMakeFiles/garbagecollector.dir/source/Collector.o:Collector.c:(.text+0x26c): first defined here
CMakeFiles/garbagecollector.dir/source/CollectorMarker.o: In function `List_removeIndex_.localalias.1':
CollectorMarker.c:(.text+0x2b3): multiple definition of `List_removeIndex_'
CMakeFiles/garbagecollector.dir/source/Collector.o:Collector.c:(.text+0x2b3): first defined here
CMakeFiles/garbagecollector.dir/source/CollectorMarker.o: In function `List_removeIndex_toIndex_':
CollectorMarker.c:(.text+0x34d): multiple definition of `List_removeIndex_toIndex_'
CMakeFiles/garbagecollector.dir/source/Collector.o:Collector.c:(.text+0x34d): first defined here
CMakeFiles/garbagecollector.dir/source/CollectorMarker.o: In function `List_remove_.localalias.0':
CollectorMarker.c:(.text+0x423): multiple definition of `List_remove_'
CMakeFiles/garbagecollector.dir/source/Collector.o:Collector.c:(.text+0x423): first defined here
CMakeFiles/garbagecollector.dir/source/CollectorMarker.o: In function `List_removeFirst_':
CollectorMarker.c:(.text+0x481): multiple definition of `List_removeFirst_'
CMakeFiles/garbagecollector.dir/source/Collector.o:Collector.c:(.text+0x481): first defined here
CMakeFiles/garbagecollector.dir/source/CollectorMarker.o: In function `List_removeLast_':
CollectorMarker.c:(.text+0x4f2): multiple definition of `List_removeLast_'
CMakeFiles/garbagecollector.dir/source/Collector.o:Collector.c:(.text+0x4f2): first defined here
CMakeFiles/garbagecollector.dir/source/CollectorMarker.o: In function `List_removeItems_':
CollectorMarker.c:(.text+0x522): multiple definition of `List_removeItems_'
CMakeFiles/garbagecollector.dir/source/Collector.o:Collector.c:(.text+0x522): first defined here
CMakeFiles/garbagecollector.dir/source/CollectorMarker.o: In function `List_at_insert_':
CollectorMarker.c:(.text+0x58e): multiple definition of `List_at_insert_'
CMakeFiles/garbagecollector.dir/source/Collector.o:Collector.c:(.text+0x58e): first defined here
CMakeFiles/garbagecollector.dir/source/CollectorMarker.o: In function `List_at_put_':
CollectorMarker.c:(.text+0x65f): multiple definition of `List_at_put_'
CMakeFiles/garbagecollector.dir/source/Collector.o:Collector.c:(.text+0x65f): first defined here
CMakeFiles/garbagecollector.dir/source/CollectorMarker.o: In function `List_swap_with_':
CollectorMarker.c:(.text+0x6c7): multiple definition of `List_swap_with_'
CMakeFiles/garbagecollector.dir/source/Collector.o:Collector.c:(.text+0x6c7): first defined here
CMakeFiles/garbagecollector.dir/source/CollectorMarker.o: In function `List_reverseInPlace':
CollectorMarker.c:(.text+0x759): multiple definition of `List_reverseInPlace'
CMakeFiles/garbagecollector.dir/source/Collector.o:Collector.c:(.text+0x759): first defined here
CMakeFiles/garbagecollector.dir/source/CollectorMarker.o: In function `List_push_':
CollectorMarker.c:(.text+0x7a6): multiple definition of `List_push_'
CMakeFiles/garbagecollector.dir/source/Collector.o:Collector.c:(.text+0x7a6): first defined here
CMakeFiles/garbagecollector.dir/source/CollectorMarker.o: In function `List_pop':
CollectorMarker.c:(.text+0x805): multiple definition of `List_pop'
CMakeFiles/garbagecollector.dir/source/Collector.o:Collector.c:(.text+0x805): first defined here
CMakeFiles/garbagecollector.dir/source/CollectorMarker.o: In function `List_top':
CollectorMarker.c:(.text+0x868): multiple definition of `List_top'
CMakeFiles/garbagecollector.dir/source/Collector.o:Collector.c:(.text+0x868): first defined here
CMakeFiles/garbagecollector.dir/source/CollectorMarker.o: In function `List_removeTrueFor_':
CollectorMarker.c:(.text+0x8a3): multiple definition of `List_removeTrueFor_'
CMakeFiles/garbagecollector.dir/source/Collector.o:Collector.c:(.text+0x8a3): first defined here
CMakeFiles/garbagecollector.dir/source/CollectorMarker.o: In function `List_qsort':
CollectorMarker.c:(.text+0x963): multiple definition of `List_qsort'
CMakeFiles/garbagecollector.dir/source/Collector.o:Collector.c:(.text+0x963): first defined here
CMakeFiles/garbagecollector.dir/source/CollectorMarker.o: In function `List_qsort_r':
CollectorMarker.c:(.text+0x999): multiple definition of `List_qsort_r'
CMakeFiles/garbagecollector.dir/source/Collector.o:Collector.c:(.text+0x999): first defined here
CMakeFiles/garbagecollector.dir/source/CollectorMarker.o: In function `List_bsearch':
CollectorMarker.c:(.text+0x9da): multiple definition of `List_bsearch'
CMakeFiles/garbagecollector.dir/source/Collector.o:Collector.c:(.text+0x9da): first defined here
CMakeFiles/garbagecollector.dir/source/CollectorMarker.o: In function `List_first':
CollectorMarker.c:(.text+0xa17): multiple definition of `List_first'
CMakeFiles/garbagecollector.dir/source/Collector.o:Collector.c:(.text+0xa17): first defined here
CMakeFiles/garbagecollector.dir/source/CollectorMarker.o: In function `List_last':
CollectorMarker.c:(.text+0xa36): multiple definition of `List_last'
CMakeFiles/garbagecollector.dir/source/Collector.o:Collector.c:(.text+0xa36): first defined here
/usr/bin/ld: cannot find -lbasekit
collect2: error: ld returned 1 exit status
CMakeFiles/garbagecollector.dir/build.make:120: recipe for target '_build/dll/libgarbagecollector.so' failed
make[2]: *** [_build/dll/libgarbagecollector.so] Error 1
CMakeFiles/Makefile2:67: recipe for target 'CMakeFiles/garbagecollector.dir/all' failed
make[1]: *** [CMakeFiles/garbagecollector.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2

Warning messages from cmake . in the garbagecollector folder include:

The dependency target "basekit" of target "garbagecollector" does not
  exist.

This also occurred with 2013.12.04.

Thanks!

--- a/libs/basekit/source/Common_inline.h
+++ b/libs/basekit/source/Common_inline.h
@@ -117,19 +117,31 @@ Kudos to Daniel A. Koepke
 	#endif 
 	
 #elif defined(__linux__) || defined(__OpenBSD__) || defined(__NetBSD__) || defined(__FreeBSD__)
-
-	#ifdef IO_IN_C_FILE
-		// in .c 
-		#define IO_DECLARE_INLINES
-		#define IOINLINE inline
-		#define IOINLINE_RECURSIVE inline
+	#ifdef __GNUC_STDC_INLINE__
+		#ifdef IO_IN_C_FILE
+			// in .c
+			#define IO_DECLARE_INLINES
+			#define IOINLINE
+			#define IOINLINE_RECURSIVE
+		#else
+			// in .h
+			#define IO_DECLARE_INLINES
+			#define IOINLINE inline
+			#define IOINLINE_RECURSIVE inline
+		#endif
 	#else
-		// in .h 
-		#define IO_DECLARE_INLINES
-		#define IOINLINE extern inline
-		#define IOINLINE_RECURSIVE extern inline
-	#endif 
-	
+		#ifdef IO_IN_C_FILE
+			// in .c
+			#define IO_DECLARE_INLINES
+			#define IOINLINE inline
+			#define IOINLINE_RECURSIVE inline
+		#else
+			// in .h
+			#define IO_DECLARE_INLINES
+			#define IOINLINE extern inline
+			#define IOINLINE_RECURSIVE extern inline
+		#endif
+	#endif
 #else
 
 	#ifdef IO_IN_C_FILE

